cmd/dep/status.go

```diff
@@ -246,6 +378,21 @@ func runStatusAll(out outputter, p *dep.Project, sm *gps.SourceMgr) error {
 				PackageCount: len(proj.Packages()),
 			}
 
+			// Get children only for specific outputers
+			// in order to avoid slower status process
+			switch out.(type) {
+			case *dotOutput:
+				r := filepath.Join(p.AbsRoot, "vendor", string(proj.Ident().ProjectRoot))
+				ptr, err := gps.ListPackages(r, string(proj.Ident().ProjectRoot))
```

Comment:
Instead of reading from `vendor` here, use a `SourceManager`'s version of `ListPackages()`. That'll take care of ensuring the files are present, unmodified, and at the right version (which you'll have to pass in). At some point, it'll also get caching for free, too.